chapter_4/ex06_pack/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# CMakeLists file for the Chapter 4 Example nr. 05: 
+# Example CMake consumer for Config-file package
+#
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.21)
+
+project(
+  ch4_ex06_pack
+  VERSION 1.0
+  DESCRIPTION "Chapter 4 Example 06, Packaging with CPack"
+  LANGUAGES CXX)
+
+if(NOT PROJECT_IS_TOP_LEVEL)
+  message(FATAL_ERROR "The chapter-4, ex06_pack project is intended to be a standalone, top-level project. Do not include this directory.")
+endif()
+
+add_subdirectory(executable)
+add_subdirectory(library)
+
+# We will not explicitly specify project name and version here and 
+# let CPack to get project name and version from the project()
+set(CPACK_PACKAGE_VENDOR "CTT Authors")
+# Enable DEB, RPM and TBZ2 generators by default
+set(CPACK_GENERATOR "DEB;RPM;TBZ2")
+# Use all available cores when parallelism is supported
+set(CPACK_THREADS 0) 
+# The DEB generator requires CPACK_DEBIAN_PACKAGE_MAINTAINER
+# value to be set.
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "CTT Authors")
```

Comment:
All of these should be specified on the CLI. I don't see a point in hardcoding them. Since this CML requests 3.21 and the reader is required to have that as well, these should be moved to a preset instead.

Reply:
It is for keeping the command short by providing defaults, so the reader can just run `cpack --config <config_file>` We've issues with space in the chapter right now and I haven't mentioned the preset files (yet) anywhere.

Reply:
I think it's a bad habit to hardcode these regardless. If you were a package maintainer and wanted to package a project that's written like this, then you would not be able to rely on the CMake documentation for the default values and now you'll have to go looking for what and where is set, then work around that. I would prefer to not cause **any** inconvenience for package maintainers, they have enough problems with existing packages already.